app/models/service.rb

```diff
@@ -528,6 +529,11 @@ def archive_as_deleted
     ::DeletedObject.create!(object: self, owner: account)
   end
 
+  def destroy_related_backend_apis
+    return if act_as_product?
+    backend_api&.destroy
```

Comment:
We should probably wait for https://github.com/3scale/porta/pull/1144 to be merged and then adapt this PR accordingly.
After that I will do a review 😄